src/InitializerTrait.php

```diff
@@ -27,12 +27,20 @@ trait InitializerTrait
     /**
      * Initialize object. Always call parent::init(). Do not call directly.
      */
-    public function init(): void
+    protected function init(): void
     {
         if ($this->_initialized) {
             throw (new Exception('Attempting to initialize twice'))
                 ->addMoreInfo('this', $this);
         }
         $this->_initialized = true;
     }
+
+    /**
+     * Do not call directly.
+     */
+    public function invokeInit(): void
```

Comment:
What is the benefit of protecting init and then exposing it with another method?

Reply:
https://github.com/atk4/core/pull/265/commits/bc9fe4a5e187ef8060411b322e4c20cd5d4d81a2

there are a lot of situation when you need before/after init method - I was thinking about hooks, but this is fine as you can override invokeInit() - this is needed when init() itself is extended, which is very ofter in ui

Reply:
Yes, I agree with that. Setting `View::$model` for instance needs to be done afterInit and not during init.
To follow common naming paterns (as in atk4/data) maybe we can use `init`, `doInit` (instead of `invokeInit`, `init`)?

Reply:
> To follow common naming paterns (as in atk4/data) maybe we can use `init`, `doInit` (instead of `invokeInit`, `init`)?

`doInit` is horrible

there are two uses and they do not invoke!, they process the operation (delete) after beforeDel and before afterDel:
![image](https://user-images.githubusercontent.com/2228672/91291405-4d39d300-e795-11ea-9793-912d94346d3e.png)

> So you keep init public but we move the actual routine under `protected doInit`. It will be no BC break then I believe

Impossible, the init method must be protected and not called directly :)

Reply:
I was talking about the https://github.com/atk4/data/pull/690 where the actual operations are performed in `doXx` methods.

> Impossible, the init method must be protected and not called directly :)

But we call directly the invokeInit? The matter is only of naming, mostly for consistency.

Reply:
doXxx there is fine in Sql sense (ie. do/send query)

here we keep init as standard method to implement

> But we call directly the invokeInit?

I understand your point - there are some uses, for easy migration, I propose to keep "some" method and keep it public, in optimal world, this method should never be called by used.